test/suites/test_socket_fd.py

```diff
@@ -0,0 +1,107 @@
+"""
+This module tests work with connection over socket fd.
+"""
+# pylint: disable=missing-class-docstring,missing-function-docstring
+
+import socket
+import sys
+import unittest
+
+import tarantool
+from .lib.skip import skip_or_run_box_session_new_tests
+from .lib.tarantool_server import TarantoolServer, find_port
+from .utils import assert_admin_success
+
+
+class TestSuiteSocketFD(unittest.TestCase):
+    EVAL_USER = "return box.session.user()"
+
+    @classmethod
+    def setUpClass(cls):
+        print(' SOCKET FD '.center(70, '='), file=sys.stderr)
+        print('-' * 70, file=sys.stderr)
+
+        cls.srv = TarantoolServer()
+        cls.srv.script = 'test/suites/box.lua'
+        cls.srv.start()
+        cls.tcp_port = find_port()
+
+        # pylint:disable=C0209
+        resp = cls.srv.admin("""
+            local socket = require('socket')
+
+            box.cfg{}
+            box.schema.user.create('test', {password = 'test', if_not_exists = true})
+            box.schema.user.grant('test', 'read,write,execute,create', 'universe',
+                                   nil, {if_not_exists = true})
+            box.schema.user.grant('guest', 'execute', 'universe',
+                                   nil, {if_not_exists = true})
+
+            socket.tcp_server('0.0.0.0', %d, function(s)
```

Comment:
The main usecase of box.session.new is plain UNIX sockets. So if we want to test full integration cycle, the scenario should be as follows.

Test suite starts a tarantool instance.
Tarantool instance creates a new session.
It also starts some small Python script with popen+inherit fds, which uses connect over fd.
Script triggers some small change that we can verify (for example, inserts a record.)
Test suite verify that the change had been applied.
(The last check is likely to be wrapped in a retry cycle.)

I'm not sure how complicated is to write such a test. The main questions are "how Tarantool instances started from unittest would resolve paths to the script" and "will script started with Tarantool succeed in package path resolve". It is likely that you will be required to write venv and no-venv version of the test.

Reply:
That's a nice update.

But I still wish to see unix socket session test.

Reply:
Oh, sorry, now I see